gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/RecommendedProductDependenciesExtension.java

```diff
@@ -30,9 +30,9 @@
     private final ProviderFactory providerFactory;
 
     @Inject
-    public RecommendedProductDependenciesExtension(ObjectFactory objectFactory, ProviderFactory providerFactory) {
-        this.recommendedProductDependencies = objectFactory.setProperty(ProductDependency.class).empty();
-        this.providerFactory = providerFactory;
+    public RecommendedProductDependenciesExtension(Project project) {
```

Comment:
Do we not have to pass in the project when we create the extension? ex: https://github.com/palantir/sls-packaging/blob/develop/gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/asset/AssetDistributionPlugin.java#L50

Reply:
we might!

Reply:
why didn't this fail originally? does gradle auto inject some things?